### PR TITLE
Upgrade Veramo and DIDComm methods

### PR DESCRIPTION
**Pull request by gbugyis**

### CODE REVIEW

File: src/components/widgets/IssueCredential.tsx [hunk 1]
```diff
@@ -101,15 +102,26 @@ const IssueCredential: React.FC<BarChartProps> = ({
...
-        data: {
-          to: subject as string,
-          from: issuer as string,
-          type: proofFormat,
-          body: proofFormat === 'jwt' ? body.proof.jwt : body,
-        },
-        save: true,
+      const messageId = uuidv4()
+      const message = {
+        type: 'application/didcomm-encrypted+json',
```
Comment (jasonhealy) [inline src/components/widgets/IssueCredential.tsx:107]: @awoie does this conflict with message type `veramo.io-chat-v1` ?

**Review by jasonhealy — APPROVED** [review]
Looks good

File: src/components/widgets/IssueCredential.tsx (hunk 1 above)
Comment (awoie) [reply src/components/widgets/IssueCredential.tsx:107]: > @awoie does this conflict with message type `veramo.io-chat-v1` ?

IMO, the packer should set the `typ` (mime-type) of the packed DIDComm message, e.g., protected header in JWE (for authcrypt). So, in that case, you should not set `application/didcomm-encrypted+json`. Also note, there is a difference between `typ` and `type`. `type` is the right one for veramo.io-chat-v1 although I'd use something like this: `veramo.io/chat/v1/basicmessage`

**Review by awoie — CHANGES_REQUESTED** [review]
media type should be set by packer

File: src/components/widgets/IssueCredential.tsx (hunk 1 above)
Comment (jasonhealy) [reply src/components/widgets/IssueCredential.tsx:107]: ok. It's the above correct for a standard non-chat message?

Comment (jasonhealy) [reply src/components/widgets/IssueCredential.tsx:107]: @gbugyis ^

Comment (awoie) [reply src/components/widgets/IssueCredential.tsx:107]: @jasonhealy doing this `type: 'application/didcomm-encrypted+json' is not correct. You'd rather set the protocol message type there, e.g., `veramo.io/chat/v1/basicmessage`.  `application/didcomm-encrypted+json` is automatically set by the packer.

Comment (gbugyis) [reply src/components/widgets/IssueCredential.tsx:107]: Gotcha. I'll update it.

File: src/components/widgets/IssueCredential.tsx [hunk 2]
```diff
@@ -101,15 +102,26 @@ const IssueCredential: React.FC<BarChartProps> = ({
...
-        data: {
-          to: subject as string,
-          from: issuer as string,
-          type: proofFormat,
-          body: proofFormat === 'jwt' ? body.proof.jwt : body,
-        },
-        save: true,
+      const messageId = uuidv4()
+      const message = {
+        type: 'veramo.io/chat/v1/basicmessagey',
```
Comment (awoie) [inline src/components/widgets/IssueCredential.tsx:107]:
```suggestion
        type: 'veramo.io/chat/v1/basicmessage',
```

**Review by awoie — APPROVED** [review]
lgtm

**Pull request merged**